test/src/main/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuilders.java

```diff
@@ -260,6 +298,25 @@ public FormLoginRequestBuilder acceptMediaType(MediaType acceptMediaType) {
 
 		private FormLoginRequestBuilder() {
 		}
+
+
+		@Override
+		public boolean isMergeEnabled() {
+			return true;
+		}
+
+		@Override
+		public Object merge( Object parent ) {
+			if (parent == null) {
+				return this;
+			}
+			if (parent instanceof MockHttpServletRequestBuilder) {
```

Comment:
I think we can just check if it is an instance of Mergeable since that is what we are casting it to. The check for what we pass in is a MockHttpServletRequestBuilder will be done when we invoke MockHttpServletRequestBuilder.merge